packages/scenario-harness/ui/app.js

```
@@ -434,6 +435,116 @@ function connectWebSocket() {
+];
+
+const ENGINE_WARNINGS_DAMAGE_MASK = 0x0080 | 0x0100;
+
+function renderEngineWarningsPanel() {
+  const container = $("engine-warnings");
+  if (!container) return;
+
+  // Render once; subsequent state changes only flip `checked`.
+  if (container.childElementCount === 0) {
+    for (const { id, label } of ENGINE_WARNINGS) {
+      const wrap = document.createElement("label");
```

ESLint

[error] 'document' is not defined.

(no-undef)